react/internal/http-switch-link.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file defines a linker middleware used for changing http verbs, allowing
+ * cache in CDNs and other http-based caching content delivery networks
+ *
+ * If the query type is Query, the request will be made using an HTTP GET
+ * If the query type is Mutation, the request will be made using HTTP POST
+ *
+ */
+import {ApolloLink} from 'apollo-link'
+import { prop, map, contains, isEmpty, isNil } from 'ramda'
+import { parse, stringify } from 'querystring'
+import { print } from 'graphql'
+
+function removeUnusedFields(obj) {
+  Object.keys(obj).forEach(key => (isNil(obj[key]) || isEmpty(obj[key])) && delete obj[key])
+
+  return obj
+}
+
+export const createHttpSwitchLink = (uri) => {
+  const [baseUrl, querystring] = uri.split('?')
```

Comment:
Splitting on ? is not a safe way to find a query. You should parse the url. Not sure what's the best way in the browser, there is probably something native

Reply:
used package `url` for this. It actually helped making the code cleaner